hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -59,89 +56,23 @@ public static EthTxData populateEthTxData(byte[] data) {
-                nonce = rlpList.get(0).asLong();
-                gasPrice = rlpList.get(1).asBytes();
-                gasLimit = rlpList.get(2).asLong();
-                to = rlpList.get(3).data();
-                value = rlpList.get(4).asBigInt();
-                callData = rlpList.get(5).data();
-                v = rlpList.get(6).asBytes();
-                BigInteger vBI = new BigInteger(1, v);
-                recId = vBI.testBit(0) ? (byte) 0 : 1;
-                r = rlpList.get(7).data();
-                s = rlpList.get(8).data();
-                if (vBI.compareTo(BigInteger.valueOf(34)) > 0) {
-                    chainId = vBI.subtract(BigInteger.valueOf(35)).shiftRight(1).toByteArray();
-                }
-            } else {
-                // typed transaction?
-                byte typeByte = rlpItem.asByte();
-                if (typeByte != 2) {
-                    // we only support EIP1559 at the moment.
-                    return null;
-                }
-                type = EthTransactionType.EIP1559;
-                rlpItem = decoder.next();
-                if (!rlpItem.isList()) {
-                    return null;
-                }
-                List<RLPItem> rlpList = rlpItem.asRLPList().elements();
-                if (rlpList.size() != 12) {
-                    return null;
-                }
-                chainId = rlpList.get(0).data();
-                nonce = rlpList.get(1).asLong();
-                maxPriorityGas = rlpList.get(2).data();
-                maxGas = rlpList.get(3).data();
-                gasLimit = rlpList.get(4).asLong();
-                to = rlpList.get(5).data();
-                value = rlpList.get(6).asBigInt();
-                callData = rlpList.get(7).data();
-                accessList = rlpList.get(8).data();
-                recId = rlpList.get(9).asByte();
-                r = rlpList.get(10).data();
-                s = rlpList.get(11).data();
+                return populateLegacyEthTxData(rlpItem, data);
             }
 
-            return new EthTxData(
-                    data,
-                    type,
-                    chainId,
-                    nonce,
-                    gasPrice,
-                    maxPriorityGas,
-                    maxGas,
-                    gasLimit,
-                    to,
-                    value,
-                    callData,
-                    accessList,
-                    recId,
-                    v,
-                    r,
-                    s);
+            // typed transaction?
+            byte typeByte = rlpItem.asByte();
+            if (typeByte != 1 && typeByte != 2) {
+                // we only support EIP-2930 and EIP-1559 type transactions at the moment.
+                return null;
+            }
+
+            if (typeByte == 1) {
+                return populateEip2390EthTxData(decoder.next(), data);
+            }
+
+            return populateEip1559EthTxData(decoder.next(), data);
```

Comment:
Use a switch expression where the default value is null and the 1 and 2 forks are the decoders.